cmd/clusterctl/cmd/config_cluster.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+type configClusterOptions struct {
+	kubeconfig             string
+	flavor                 string
+	bootstrapProvider      string
+	infrastructureProvider string
+
+	targetNamespace   string
+	kubernetesVersion string
+	controlplaneCount int
+	workerCount       int
+}
+
+var cc = &configClusterOptions{}
+
+var configClusterClusterCmd = &cobra.Command{
+	Use:   "cluster",
+	Short: "Generate templates for creating a Cluster API workload clusters",
+	Long: LongDesc(`
+		clusterctl ships with a list of well know providers; if necessary, edit
+		the $HOME/.clusterctl.yaml file to add new provider configurations or to customize existing ones.
+		
+		Each provider configuration links to a repository, and clusterctl will fetch the template
+		from creating a new cluster from there.`),
+
+	Example: Examples(`
+		# Generates a yaml file for for creating a Cluster API workload clusters using
+		# default infrastructure provider and default bootstrap provider installed in the cluster.
+		clusterctl config cluster my-cluster
```

Comment:
nit: can we surround values we know are parameters with <>? 
```
clusterctl config cluster <my-cluster>
```

Reply:
(same for the rest of the usage strings)

Reply:
I'm fine surrounding values in the command usage, but examples IMO should be like the real commands. Ideally, you can copy-paste them and run the command without any change

Reply:
> Ideally, you can copy-paste them and run the command without any change

Good point, let's leave this as-is